include/DataFile.h

```diff
@@ -71,7 +72,9 @@ class LMMS_EXPORT DataFile : public QDomDocument
 	QString nameWithExtension( const QString& fn ) const;
 
 	void write( QTextStream& strm );
-	bool writeFile( const QString& fn );
+	bool writeFile(const QString& fn, bool withResources = false);
```

Comment:
Since writeFile receives the file name anyways, can you just pass it over to `copyResources` instead of introducing `m_filename` for that? Are these two variables always the same?

Reply:
The file name `writeFile` receives as an argument is the destination file, `m_filename` on the other hand is the file that the `DataFile` object originated from, which is used to resolve the local paths.

Reply:
I'm not less confused then before your answer :laughing:

Reply:
`writeFile` writes a `DataFile` object (i.e.: project file XML) to a file (used to save the project), the file it receives as a parameter is the file it will write it to. But a `DataFile` can be constructed from a file as well, which happens when we load a project from a file (`Song::loadProject` and also on `main.cpp` in the `makebundle` or `upgrade` actions for example). That file it originated from is what's stored in `m_filename`, if the `DataFile` didn't originate from a file `m_filename` is an empty string.

`copyResources` might need to resolve the local paths itself if we are running the command from CLI, so it uses `m_filename` to do so. It couldn't use the filename received as an argument to `writeFile` because that would be the file it's saving the bundle **to**, not the file that it's loading the content **from**.

Reply:
Got it now. Like discussed in PM, `m_filename` is used to *read* the resources before writing copies of them. So we need the origin of the DataFile. Resolving my comment.